model/host/hostutil.go

```diff
@@ -1038,6 +1046,74 @@ func (h *Host) SpawnHostSetupCommands(settings *evergreen.Settings) (string, err
 	return h.spawnHostSetupConfigDirCommands(conf), nil
 }
 
+// PopulateGithubToken adds a read-only token for owner/repo associated with the task
+// to h.ProvisionOptions
+func (h *Host) PopulateGithubToken(ctx context.Context) error {
+	p := h.ProvisionOptions
+	if p.TaskId == "" {
+		return errors.New("missing task id")
+	}
+	if p.FetchOpts == nil || p.FetchOpts.ProjectOwner == "" || p.FetchOpts.ProjectRepo == "" {
+		grip.Warning(message.Fields{
+			"message": "missing project owner or repo when populating token for spawn host",
```

Comment:
I mostly added it because there isn't much visibility into fetch failing as it runs in memory on the Jasper service, which doesn't record stdout/stderr logs. So the guidance is usually to just run fetch again. I was trying to add logs where I can. 
But I see your point. I'll remove them.